src/lints/function_export_name_changed.ron

```diff
@@ -0,0 +1,47 @@
+SemverQuery(
+    id: "function_export_name_changed",
+    human_readable_name: "function's export name has changed or been removed",
+    description: "A function's ABI name with #[no_mangle] or #[export_name = \"name\"] has changed or been removed",
+    required_update: Major,  // Not specified in rust semver doc? most similar is for #[repr] changes: https://doc.rust-lang.org/cargo/reference/semver.html#type-layout
+    reference_link: Some("https://doc.rust-lang.org/reference/abi.html#the-no_mangle-attribute"),  // TODO
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Function {
+                        importable_path {
+                            path @output @tag
+                        }
+
+                        export_name @output @tag @filter(op: "is_not_null")
+                    }
+                }
+            }
+            
+            current {
+                item {
+                    ... on Function {
+                        name @output
+
+                        export_name @filter(op: "!=", value: ["%export_name"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+    },
+    error_message: "the ABI export name of a function has been changed or been removed through the #[no_mangle] or #[export_name] attributes",
+    per_result_error_template: Some("function {{join \"::\" path}} in file {{span_filename}}:{{span_begin_line}} with old export name {{export_name}}"),
```

Comment:
The `span_filename` is a full path, which may be quite long, so I would recommend keeping the span info as the last thing in the template -- anything after it is likely to be missed.

Perhaps something like this:
```suggestion
    per_result_error_template: Some("function {{join \"::\" path}} no longer has ABI name {{export_name}} in {{span_filename}}:{{span_begin_line}}"),
```